conans/client/build/autotools_environment.py

```diff
@@ -144,39 +144,58 @@ def configure(self, configure_dir=None, args=None, build=None, host=None, target
             pkg_env = {"PKG_CONFIG_PATH": [self._conanfile.install_folder]} \
                 if "pkg_config" in self._conanfile.generators else {}
 
+        configure_dir = self._adjust_path(configure_dir)
+
         if self._conanfile.package_folder is not None:
             if not args:
                 args = ["--prefix=%s" % self._conanfile.package_folder.replace("\\", "/")]
             elif not self._is_flag_in_args("prefix", args):
                 args.append("--prefix=%s" % self._conanfile.package_folder.replace("\\", "/"))
 
+            all_flags = ["bindir", "sbin", "libexec", "libdir", "includedir", "oldincludedir",
+                         "datarootdir"]
+            help_output = self._help_output(configure_dir)
+            not_available = [flag for flag in all_flags if "--%s" % flag not in help_output]
```

Comment:
What if the output is something like...?

```
 --libexecute use this parameter instead of --libexec because we decided not to be standard guys...
```

I'll be a little bit more strict in this check, maybe checking that the line starts with the parameter